sklearn/neighbors/base.py

```diff
@@ -246,8 +248,10 @@ def _fit(self, X):
                     self.metric != 'precomputed'):
                 if self.effective_metric_ in VALID_METRICS['kd_tree']:
                     self._fit_method = 'kd_tree'
-                else:
+                elif self.effective_metric_ in VALID_METRICS['ball_tree']:
```

Comment:
Same comment as above. You should maybe add a line to the existing `test_callable_metric` to check that the underlying fit is the ball_tree.